src/main/scala/performanceanalysis/logreceiver/LogReceiverActor.scala

```diff
@@ -13,28 +13,31 @@ object LogReceiverActor {
 }
 
 class LogReceiverActor extends Actor with ActorLogging {
-
-  def receive: Receive = normal(Map.empty[String, ActorRef])
-
-  def normal(logParserActors: Map[String, ActorRef]): Receive = {
-    case SubmitLogs(componentId, logs) =>
-      handleSubmitLogs(logParserActors, componentId, logs)
-    case RegisterNewLogParser(componentName, newLogParser) =>
-      handleNewLogParser(logParserActors, componentName, newLogParser)
+  private type ComponentId = String
+  private type LogParser = ActorRef
+  private type LogParsers = Map[ComponentId, LogParser]
```

Comment:
Agree, `LogParsersById` is a bit more clear (and prevents further discussion on this ;)